deebot_client/map.py

```diff
@@ -314,74 +417,103 @@ def get_base64_map(self, width: int | None = None) -> bytes:
-        for subset in self._map_data.map_subsets.values():
-            _draw_subset(subset, draw, image_box)
+            _LOGGER.debug("[get_svg_map] Crop Image")
+            cropped = image.crop(image_box)
+            del image
 
-        del draw
+            _LOGGER.debug(
+                "[get_svg_map] Map current Size: X: %d Y: %d",
+                cropped.size[0],
+                cropped.size[1],
+            )
 
-        _draw_positions(self._map_data.positions, image, image_box)
+            _LOGGER.debug("[get_svg_map] Saving to buffer")
+            buffered = BytesIO()
+            cropped.save(buffered, format="PNG")
+            del cropped
 
-        _LOGGER.debug("[get_base64_map] Crop Image")
-        cropped = image.crop(image_box)
-        del image
+            base64_bg = base64.b64encode(buffered.getvalue())
 
-        _LOGGER.debug("[get_base64_map] Flipping Image")
-        cropped = ImageOps.flip(cropped)
+            # Build the SVG elements
 
-        _LOGGER.debug(
-            "[get_base64_map] Map current Size: X: %d Y: %d",
-            cropped.size[0],
-            cropped.size[1],
-        )
+            svg_positions = _get_svg_positions(self._map_data.positions, image_box)
 
-        new_size = None
-        if width is not None and width > 0:
-            height = int((width / cropped.size[0]) * cropped.size[1])
-            _LOGGER.debug(
-                "[get_base64_map] Resize based on the requested width: %d and calculated height %d",
-                width,
-                height,
+            svg_subset_elements: list[svg.Element] = [
+                _get_svg_subset(subset, image_box)
+                for subset in self._map_data.map_subsets.values()
+            ]
+
+            svg_traces_path = self._get_svg_traces_path()
+
+            # Elements of the SVG Map to vertically flip
+            svg_map_group_elements: list[svg.Element] = []
+
+            # Map background.
+            svg_map_group_elements.append(
+                svg.Image(
+                    x=image_box[0],
+                    y=image_box[1],
+                    width=image_box[2] - image_box[0],
+                    height=image_box[3] - image_box[1],
+                    style="image-rendering: pixelated",
+                    href=f"data:image/png;base64,{base64_bg.decode('ascii')}",
+                )
             )
-            new_size = (width, height)
-        elif cropped.size[0] > 400 or cropped.size[1] > 400:
-            _LOGGER.debug("[get_base64_map] Resize disabled.. map over 400")
-        else:
-            resize_factor = Map.RESIZE_FACTOR
-            _LOGGER.debug("[get_base64_map] Resize factor: %d", resize_factor)
-            new_size = (
-                cropped.size[0] * resize_factor,
-                cropped.size[1] * resize_factor,
+
+            # Additional subsets (VirtualWalls and NoMopZones)
+            svg_map_group_elements.extend(svg_subset_elements)
```

Comment:
```suggestion
            svg_map_group_elements.extend(_get_svg_positions(self._map_data.positions, image_box))
```

Reply:
Having bot and station svg elements defined before the path will make them appear "below" the path itself. Is it the intended result?

Reply:
There was a copy-paste error, and thanks for catching it and asking :)

I wanted only to remove the variables above as we can calculate the SVG elements when we use it. I have updated the suggestions